registry.go

```diff
@@ -0,0 +1,114 @@
+package manet
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	ma "github.com/jbenet/go-multiaddr"
+)
+
+type AddrParser func(a net.Addr) (ma.Multiaddr, error)
+type MaddrParser func(ma ma.Multiaddr) (net.Addr, error)
```

Comment:
instead of these two functions, which are named a bit confusingly, either:

```go
type NetAddr2MAddr func(a net.Addr) (ma.Multiaddr, error)
type MAddr2NetAddr func(ma ma.Multiaddr) (net.Addr, error)

// or
type FromNetAddr func(a net.Addr) (ma.Multiaddr, error)
type ToNetAddr func(ma ma.Multiaddr) (net.Addr, error)

// or even
type NetCodec struct {
    FromNet func(a net.Addr) (ma.Multiaddr, error)
    ToNet   func(a net.Addr) (ma.Multiaddr, error)
}

var TCPNetCodec = NetCodec{}
TCPNetCodec.FromNet = func ...
TCPNetCodec.ToNet = func ...
```